src/Compilers/CSharp/Portable/Symbols/Source/ExplicitInterfaceHelpers.cs

```diff
@@ -220,7 +220,7 @@ public static string GetMemberNameWithoutInterfaceName(string fullName)
                 // At this point, we know that explicitInterfaceNamedType is an interface, so candidate must be public
                 // and, therefore, accessible.  So we don't need to check that.
```

Comment:
> It would appear that an access check is now required here.

The impact of accessibility on this scenario hasn't been discussed yet at LDM and my understanding is that it is already on the list of issues to look at. For now, I'll replace the relevant portion of the original comment with a PROTOTYPE comment and leave behavior as is.